flake.nix

```diff
@@ -19,25 +19,43 @@
         haskell-backend.overlay
         (final: prev:
           let
-            version =
+            kore-version =
               prev.haskell-backend-stackProject.hsPkgs.kore.components.exes.kore-exec.version;
             kore = prev.symlinkJoin {
-              name = "kore-${version}-${haskell-backend.sourceInfo.shortRev or "local"}";
+              name = "kore-${kore-version}-${
+                  haskell-backend.sourceInfo.shortRev or "local"
+                }";
               paths = prev.lib.attrValues
                 prev.haskell-backend-stackProject.hsPkgs.kore.components.exes;
             };
-
+            k-version =
+              prev.lib.removeSuffix "\n" (builtins.readFile ./package/version);
             src = prev.stdenv.mkDerivation {
-              name = "llvm-source";
+              name = "k-${k-version}-${self.rev or "dirty"}-src";
```

Comment:
when nix builds this derivation, it checks if the current folder, which is a git repo has been modified. if not, `self.rev` will point to the current git commit hash, otherwise it is empty so the `or "dirty"` will substitute in "dirty" instead...